drivers/bluetooth/hci/hci_spi_st.c

```diff
@@ -334,13 +334,27 @@ static int bt_spi_bluenrg_setup(const struct bt_hci_setup_params *params)
 }
 #endif /* CONFIG_BT_BLUENRG_ACI */
 
-static struct net_buf *bt_spi_rx_buf_construct(uint8_t *msg)
+static int bt_spi_rx_buf_construct(uint8_t *msg, struct net_buf **bufp, uint16_t size)
 {
 	bool discardable = false;
 	k_timeout_t timeout = K_FOREVER;
 	struct bt_hci_acl_hdr acl_hdr;
-	struct net_buf *buf;
-	int len;
+	static int len;
```

Comment:
I was scratching my head around some of the new code, until I realised you're transforming `len` into a static variable. Maybe add a code comment here or rename it to something more descriptive (that gives a better hint at its persistency)? Alternatively, could the same be solved by moving it out into the loop of the calling function (like you're doing with `bufp`)?

Reply:
I've added a descriptive comment for `len`.
Alternative solution requires more changes on the code.